tests/unit/test_adjoints.py

```diff
@@ -0,0 +1,218 @@
+import pytest
+import pickle
+from firedrake import *
+from firedrake.adjoint import *
+from checkpoint_schedules import SingleMemoryStorageSchedule, SingleDiskStorageSchedule
+import numpy as np
+from pathlib import Path
+
+
+def tape_generation_staggered_solves(scheduler):
+
+    continue_annotation()
+
+    # get tape
+    tape = get_working_tape()
+    tape.clear_tape()
+    ret = {}
+
+    if scheduler is not None:
+        if isinstance(scheduler, SingleDiskStorageSchedule):
```

Comment:
I think we can use `schedule.uses_storage_type(StorageType.DISK)` to be a bit more generic.